nomad/acl.go

```diff
@@ -161,9 +161,14 @@ func (s *Server) remoteIPFromRPCContext(ctx *RPCContext) (net.IP, error) {
 // for the identity they intend the operation to be performed with.
 func (s *Server) ResolveACL(args structs.RequestWithIdentity) (*acl.ACL, error) {
 	identity := args.GetIdentity()
-	if !s.config.ACLEnabled || identity == nil {
+	if !s.config.ACLEnabled {
 		return nil, nil
```

Comment:
OT: I don't think we should do this for this PR, but I am curious if we could get away from using `nil` as a sentinel value by switching these helpers to return a management token:

```suggestion
	if !s.config.ACLEnabled {
		return structs.ManagementToken(), nil
```

Maybe I'll find an opportunity to poke at that more in my WID work... :thinking:

Reply:
Good call. As per our sidebar discussion, I've backed out the changes to this file in this PR